ccmlib/common.py

```diff
@@ -76,6 +77,36 @@ class UnavailableSocketError(CCMError):
     pass
 
 
+class LogPatternToVersion(object):
+
+    def __init__(self, versions_to_patterns):
+        self.versions_to_patterns = versions_to_patterns
+
+    def __call__(self, version):
+        pattern = None
+        for key in sorted(self.versions):
+                if version >= key:
+                    pattern = self.versions_to_patterns[key]
```

Comment:
Still not sure about this loop, sorry -- it's just kinda hard to reason about. I think a more readable strategy here would be something like

```python
greatest_key_less_than_version = max(k for k in self.versions_to_patterns if k < version)
return self.versions_to_patterns[greatest_key_less_than_version]
```

The other thing to be concerned about is that we should somehow make sure that we compare `LooseVersion`s here. What API do we want? I'm tempted to have logic like

`LooseVersion(v) if not isinstance(v, LooseVersion) else v`. @ptnapoleon, thoughts?